server/service/src/processors/transfer/requisition/create_response_requisition.rs

```diff
@@ -0,0 +1,137 @@
+use crate::{number::next_number, requisition::common::get_lines_for_requisition};
+
+use super::{RequisitionTransferProcessor, RequisitionTransferProcessorRecord};
+use chrono::Utc;
+use repository::{
+    NumberRowType, RepositoryError, Requisition, RequisitionLineRow, RequisitionLineRowRepository,
+    RequisitionRow, RequisitionRowRepository, RequisitionRowStatus, RequisitionRowType,
+    StorageConnection,
+};
+use util::uuid::uuid;
+
+pub struct CreateResponseRequisitionProcessor;
+impl RequisitionTransferProcessor for CreateResponseRequisitionProcessor {
+    fn get_description(&self) -> String {
+        "Create response requisition from request requisition".to_string()
+    }
+
+    fn try_process_record(
+        &self,
+        connection: &StorageConnection,
+        record_for_processing: &RequisitionTransferProcessorRecord,
+    ) -> Result<Option<String>, RepositoryError> {
+        // Check can execute
+        let RequisitionTransferProcessorRecord {
+            linked_requisition,
+            requisition: source_requisition,
+            ..
+        } = &record_for_processing;
+
+        if source_requisition.requisition_row.r#type != RequisitionRowType::Request {
+            return Ok(None);
+        }
+
+        if source_requisition.requisition_row.status != RequisitionRowStatus::Sent {
+            return Ok(None);
+        }
+
+        if linked_requisition.is_some() {
+            return Ok(None);
+        }
+
+        // Execute
+        let new_requisition =
+            generate_linked_requisition(connection, &source_requisition, record_for_processing)?;
+
+        let new_requisition_lines = generate_linked_requisition_lines(
+            connection,
+            &new_requisition,
+            &source_requisition.requisition_row,
+        )?;
+
+        RequisitionRowRepository::new(connection).upsert_one(&new_requisition)?;
+
+        let requisition_line_row_repository = RequisitionLineRowRepository::new(connection);
+
+        for line in new_requisition_lines.iter() {
+            requisition_line_row_repository.upsert_one(line)?;
+        }
+
+        let result = format!(
+            "requisition ({}) lines ({:?}) source requisition ({})",
+            new_requisition.id,
+            new_requisition_lines.into_iter().map(|r| r.id),
+            source_requisition.requisition_row.id
+        );
+
+        Ok(Some(result))
+    }
+}
+
+fn generate_linked_requisition(
+    connection: &StorageConnection,
+    source_requisition: &Requisition,
+    record_for_processing: &RequisitionTransferProcessorRecord,
+) -> Result<RequisitionRow, RepositoryError> {
+    let store_id = record_for_processing.other_party_store_id.clone();
+    let name_id = source_requisition.store_row.name_id.clone();
+
+    let source_requistition_row = &source_requisition.requisition_row;
+
+    let result = RequisitionRow {
```

Comment:
![Screen Shot 2022-09-13 at 7 18 30 PM](https://user-images.githubusercontent.com/26194949/189836115-6275ffe7-8bfd-4287-b418-45d48f3cc560.png)

We use it quite a bit. I tend to agree that we should be a bit more descriptive. I've been using this syntax quite a lot, to avoid extra nesting with Ok(), just makes it look slightly cleaner. I hope it's not a biggy since it's quite convenient to not think too much about naming if the variable is returned straight away with `Ok(result)`. I am happy to reconsider and use more descriptive naming if it proves to be a problem.

Reply:
Not a major at all, I use it a LOT too!
Sorry I think I've slipped into using these PR reviews to discuss/reflect on stylistic things.
In this case when reading the I found myself asking why is this a `result` when it's just creating a struct?
It was actually how I noticed that the number creation could fail during struct creation, so the variable naming kind of helped me in a way.

Reply:
> Sorry I think I've slipped into using these PR reviews to discuss/reflect on stylistic things. 

Can have a chat in person sometime, and we can summaries in omSupply chat